### DISCUSSION TITLE
Multiple mantine provider on a page

### DISCUSSION THREAD

**Discussion by Canic** [mantinedev/mantine#6596] [Q&A]
Hey,

I have a small question that I couldn't find a good answer for.

We are using Mantine in a hybrid setup (Remix SSR with WordPress as a proxy). We inject Mantine multiple times. Is it acceptable to use the Mantine provider multiple times within new component indexes on a page?

Or is there a recommended way to manage different Mantine apps at various positions on the page?

We primarily use Mantine as a web component.

**Comment by rtivital** [comment]
It is not recommended to have multiple MantineProviders in your application

**Reply by screenwork** [reply]
We have a similar problem and aren't sure how to solve it without using multiple MantineProviders. That's the main reason why we're using shadcn instead of doing everything with Mantine.

We have an application where we output several web components. Each web component is rendered individually as a React application and therefore needs its own MantineProvider.

Unfortunately, we can't use Portals because we're partially rendering with server-side rendering (SSR), which would complicate everything – at least that's what I think. It would be simpler if we could use the MantineProvider multiple times. ;-)
